### PR TITLE
Fix for atlassian OAuth flow

### PR DESCRIPTION
**Pull request by swnia**
## Hey, I just made a Pull Request!

As described in #23031 and documented [here](https://developer.atlassian.com/cloud/jira/platform/oauth-2-3lo-apps/#1--direct-the-user-to-the-authorization-url-to-get-an-authorization-code) the Atlassian auth provider targets the wrong endpoint and has the wrong default audience.
This PR fixes that.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hi! This needs a changeset (run `yarn changeset` and follow the instructions) to get released.

Is there no risk of this being a difference between like different atlassian offerings? Sorry for the open ended question, but we've sometimes seen something fix a cloud version of an offering but breaking an on-premise version :)

**Comment by swnia** [comment]
For [JIRA-server](https://confluence.atlassian.com/adminjiraserver/jira-oauth-2-0-provider-api-1115659070.html) there is never an /oauth/ before the keyword. So, this never worked for JIRA-server and the change does not improve on that.

**Comment by postsa** [comment]
+1 on this change. We just downgraded to `0.20.0` to fix an auth issue introduced by the `plugin-auth-backend-module-atlassian-provider`. The working version uses `https://auth.atlassian.com/authorize` https://github.com/backstage/backstage/blob/4a111890893884b28991ab09d3adacdf88811113/plugins/auth-backend/src/providers/atlassian/strategy.ts#L52.

File: plugins/auth-backend-module-atlassian-provider/src/authenticator.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export const atlassianAuthenticator = createOAuthAuthenticator({
     const clientId = config.getString('clientId');
     const clientSecret = config.getString('clientSecret');
     const baseUrl =
-      config.getOptionalString('audience') || 'https://atlassian.com';
+      config.getOptionalString('audience') || 'https://api.atlassian.com';
```
[the remark this answers is not in the document]
Comment (swnia) [reply plugins/auth-backend-module-atlassian-provider/src/authenticator.ts:33]: Screenshot 2024-03-01 at 10 19 23
The docs say to set the audience to `api.atlassian.com` which is why I made the change.
The audience should not be the baseUrl, though. I'll update that.

**Review by postsa — APPROVED** [review]
Looks good to me. The auth url is not configurable at the moment, but it seems like there is only one correct value so that's probably fine. Although making it configurable might be enough to solve for the on-prem case.

File: plugins/auth-backend-module-atlassian-provider/src/authenticator.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (postsa) [reply plugins/auth-backend-module-atlassian-provider/src/authenticator.ts]: Hmmm, given that the default is correct, can it just be 

```
config.getOptionalString('host')
```

that allows for `passport-atlassian-oauth2` to set the correct default value and provides configuration for on prem jira.

@freben - Would there be a more idiomatic config property name?

[the remark this answers is not in the document]
Comment (swnia) [reply plugins/auth-backend-module-atlassian-provider/src/authenticator.ts]: As linked above, on-prem JIRA server has different paths according to what I could find. So, the plugin would need to allow to configure all of these values.

**Comment by swnia** [comment]
@benjdlambert @camilaibs is something missing in the PR such that it could get merged?

**Review by Rugvip — APPROVED** [review]
Alright, looks good, thank you! 👍

**Pull request merged**